src/plugins/workspace/public/application.tsx

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { AppMountParameters, ScopedHistory } from '../../../core/public';
+import { OpenSearchDashboardsContextProvider } from '../../opensearch_dashboards_react/public';
+import { WorkspaceFatalError } from './components/workspace_fatal_error';
+import { Services } from './types';
+
+export const renderFatalErrorApp = (params: AppMountParameters, services: Services) => {
+  const { element } = params;
+  const history = params.history as ScopedHistory<{ error?: string }>;
+  ReactDOM.render(
+    <OpenSearchDashboardsContextProvider services={services}>
+      <WorkspaceFatalError error={history.location.state.error} />
```

Comment:
if error is null, should we consider `const error = history.location.state?.error;` to ensure error exists before accessing it, then it won't throw an error when rendering the component.

Reply:
Sure, done for that.